### ISSUE TITLE
Unable to authorise client

### ISSUE THREAD

**Issue by Auroray** [ekutner/home-connect-hass#57]
Hi,

I've tried setting up this integration out of curiosity to see how it compares with the one included in HA Core - so far I wasn't able to authorise it against the new app I created for it in HC Developer portal. I've seen the one already closed issue where the redirect was incorrect, but in my case the App ID, client secret and redirect URL are correct, and the BSH API still says no.
<img width="847" alt="HCA-2" src="https://user-images.githubusercontent.com/26858980/174442117-2cdae253-4757-4f37-b9a4-76309a437087.png">
<img width="852" alt="HCA-1" src="https://user-images.githubusercontent.com/26858980/174442116-bb42e1bb-40f2-441f-8b24-464d6c0fc2e6.png">
<img width="847" alt="HCA-3" src="https://user-images.githubusercontent.com/26858980/174442154-7ee04db3-96aa-4bf6-979b-bc10a2b45516.png">

Any ideas?

**Comment by ekutner** [comment]
Did you configure my.home-assistant.io to point to your HA instance? If not then you need to go to that URL in your browser and do it there.

**Comment by infabo** [comment]
Is it possible to use the callback url of the local instance instead of the cloud url (my.home-assistant.io)? I don't know where this comes from. I do not have an cloud acc.

**Comment by ekutner** [comment]
No. It's a HA thing, unrelated to the integration and you don't need a cloud account to use it.

**Comment by Auroray** [comment]
Alright, I was able to resolve it but ho idea how. I confirmed the issue was definitely related to the callback URL, but after modifying it about 100 times both in the app and in the authorisation request URL it just finally worked, so I've got no clues for anybody that enounters this issue again. One thing I noticed was that sometimes the authorisation URL contained one more parameter - redirect URL. Not sure of the significance of this.
